src/cffi.zig

```diff
@@ -254,8 +254,20 @@ export fn bb_import_package_deinit(package: ?*ModuleImportPackage) void {
     }
 }
 
-export fn bb_import_package_add_function(package: ?*ModuleImportPackage, func: ?CHostFunction, c_name: ?[*:0]const u8, c_params: ?[*]ValType, num_params: usize, c_returns: ?[*]ValType, num_returns: usize, userdata: ?*anyopaque) CError {
-    if (package != null and c_name != null and func != null) {
+
+const HostFunc = extern struct {
+    userdata: ?*anyopaque,
+    callback: CHostFunction,
+};
+
+fn trampoline(userdata: ?*anyopaque, module: *core.ModuleInstance, params: [*]const Val, returns: [*]Val) error{}!void {
+    const host: *HostFunc = @alignCast(@ptrCast(userdata));
+
+    @call(.auto, host.callback, .{host.userdata, module, params, returns});
+}
+
+export fn bb_import_package_add_function(package: ?*ModuleImportPackage, c_name: ?[*:0]const u8, c_params: ?[*]ValType, num_params: usize, c_returns: ?[*]ValType, num_returns: usize, userdata: ?*HostFunc) CError {
```

Comment:
minor: would prefer `userdata` gets renamed to `func` and moved to the original parameter location

Reply:
same for `bb_import_package_add_function` in the C header